js/interface.js

```diff
@@ -181,11 +179,17 @@ $('.show-email-provider').on('click', function() {
   emailProvider = Fliplet.Widget.open('com.fliplet.email-provider', {
     data: emailProviderData
   });
+  Fliplet.Studio.emit('widget-save-label-update', {
+    text: 'Save'
+  });
 
   emailProvider.then(function onForwardEmailProvider(result) {
     emailProvider = null;
-    emailProviderResult = result.data;
+    dataSource.definition.validation.email.template = result.data;
```

Comment:
Can you elaborate on why these changes are required? I'm confused at why this is necessary. Furthermore, it might fail if one of the properties is missing? e.g. https://github.com/Fliplet/fliplet-widget-validation-manager/blob/0590bda8f1142bb1f28f863954916011823348bd/js/interface.js#L167

Reply:
@squallstar 
At line 167 there is a check that looks into dataSource and decides to use data form dataSource or default template. If there is no template in dataSource we use the default template and on the contrary when dataSource has the template. I have discovered that dataSource updated only when you press save in the main frame and that's why we can see only template email subject when trying to go back before save the main frame. Therefore I decided to update dataSource after closing this provider.
And yes I can see now that not all dataSource has needed properties. Maybe if I add a check that looks if there are all needed properties in the dataSource object and will add them if they are not, will it solve the issue?

Reply:
@YaroslavOvdii looks like you haven't take any action on this yet?